include/ck_tile/ops/gemm/pipeline/block_gemm_pipeline_agmem_bgmem_creg_v1.hpp

```diff
@@ -140,8 +150,7 @@ struct BlockGemmPipelineAGmemBGmemCRegV1
         }
 
         index_t iCounter = num_loop - 1;
-
-        do
+        while(iCounter > 0)
```

Comment:
Why swap from `do-while` to `while-do`? Technically, `do-while` should deliver better performance due to less branches.

Reply:
Because in the do-while loop it if we only have the num_loop = 1 and the iCounter is 0. It will execute this block of the code. However, it should skip that and go to the tile block directly.